posthog/api/organization_member.py

```diff
@@ -71,10 +73,24 @@ class OrganizationMemberViewSet(
 ):
     serializer_class = OrganizationMemberSerializer
     permission_classes = [IsAuthenticated, OrganizationMemberPermissions, OrganizationMemberObjectPermissions]
-    queryset = OrganizationMembership.objects.exclude(user__email__endswith=INTERNAL_BOT_EMAIL_SUFFIX)
+    queryset = OrganizationMembership.objects.all().order_by(Lower('user__first_name')).exclude(user__email__endswith=INTERNAL_BOT_EMAIL_SUFFIX)
     lookup_field = "user__uuid"
     ordering = ["level", "-joined_at"]
 
+    def list(self, request, *args, **kwargs):
+        #put the loggedInUser object on top as per issue:8234
+        queryset = self.filter_queryset(self.get_queryset())
+        loggedInUser  = queryset.filter(user = request.user).first()
+        queryset = queryset.exclude(user = request.user)
+        queryset = [loggedInUser] + list(queryset)
+        page = self.paginate_queryset(queryset)
+        if page is not None:
+            serializer = self.get_serializer(page, many=True)
+            return self.get_paginated_response(serializer.data)
+
+        serializer = self.get_serializer(queryset, many=True)
+        return Response(serializer.data)
```

Comment:
You don't have to move this away from the endpoint then.
However, if keeping this in the backend, the sorting logic should sit entirely in `get_queryset` instead of `list`, as we don't want to interfere with pagination or serialization at all.
Also, it's not guaranteed forever that the logged-in user should be part of the queryset – there may be a case where a superuser wants to list users of an organization they themselves don't belong to. Instead, the "current user comes first" logic should be part of the ORDER BY clause, e.g. by using `.order_by(Case(When(user__id=request.user.id, then=0), default=1))`.

Reply:
Thanks @Twixes I will look into this and try to improve the logic and handle the cases